<a href="https://colab.research.google.com/github/SURESHASWATHI/assignment/blob/main/basicstat2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
durability_data = [1.1, 1.3, 1.0, 1.2, 1.1, 1.3, 1.2, 1.1, 1.0, 1.2]

In [2]:
import numpy as np

sample_mean_a = np.mean(durability_data)
sample_std_dev_a = np.std(durability_data, ddof=1)

print(f"Sample Mean: {sample_mean_a}")
print(f"Sample Standard Deviation: {sample_std_dev_a}")

Sample Mean: 1.15
Sample Standard Deviation: 0.10801234497346433


**Reasoning**:
Calculate the degrees of freedom and the t-score for a 99% confidence interval.



In [3]:
degrees_of_freedom = len(durability_data) - 1
from scipy.stats import t
alpha = 1 - 0.99
lower_tail_prob = alpha / 2
t_score_a = abs(t.ppf(lower_tail_prob, degrees_of_freedom))
print(f"Degrees of Freedom: {degrees_of_freedom}")
print(f"T-score for 99% confidence interval: {t_score_a}")

Degrees of Freedom: 9
T-score for 99% confidence interval: 3.2498355415921254


**Reasoning**:
Calculate the standard error and then the margin of error using the provided variables and store it in `margin_of_error_a`.



In [4]:
import math

sample_size = len(durability_data)
standard_error_a = sample_std_dev_a / math.sqrt(sample_size)
margin_of_error_a = t_score_a * standard_error_a

print(f"Standard Error (Scenario A): {standard_error_a}")
print(f"Margin of Error (Scenario A): {margin_of_error_a}")

Standard Error (Scenario A): 0.03415650255319866
Margin of Error (Scenario A): 0.11100301597386718


**Reasoning**:
Calculate the lower and upper bounds of the confidence interval and print the result.



In [5]:
lower_bound_a = sample_mean_a - margin_of_error_a
upper_bound_a = sample_mean_a + margin_of_error_a

print(f"99% Confidence Interval (Scenario A): [{lower_bound_a:.3f}, {upper_bound_a:.3f}]")

99% Confidence Interval (Scenario A): [1.039, 1.261]


**Reasoning**:
Explain why the t-distribution is appropriate for constructing the confidence interval in Scenario A, mentioning the conditions for using the t-distribution and relating them to Scenario A.



In [6]:
print("Rationale for using the t-distribution in Scenario A:")
print("The t-distribution is used for constructing confidence intervals for the mean when the sample size is small AND the population standard deviation is unknown. In Scenario A, we have a small sample size (n=10) and we are using the sample standard deviation as an estimate because the population standard deviation is not provided or assumed to be unknown. Under these conditions, the t-distribution provides a more appropriate and conservative estimate for the confidence interval compared to the z-distribution, which assumes a known population standard deviation or a large sample size (typically n > 30). The t-distribution accounts for the additional uncertainty introduced by estimating the population standard deviation from the sample.")

Rationale for using the t-distribution in Scenario A:
The t-distribution is used for constructing confidence intervals for the mean when the sample size is small AND the population standard deviation is unknown. In Scenario A, we have a small sample size (n=10) and we are using the sample standard deviation as an estimate because the population standard deviation is not provided or assumed to be unknown. Under these conditions, the t-distribution provides a more appropriate and conservative estimate for the confidence interval compared to the z-distribution, which assumes a known population standard deviation or a large sample size (typically n > 30). The t-distribution accounts for the additional uncertainty introduced by estimating the population standard deviation from the sample.


**Reasoning**:
The user wants me to acknowledge that the sample mean is already calculated and stored in the `sample_mean_a` variable. I will print a message to confirm this and then finish the task.



In [7]:
print("The sample mean has already been calculated and stored in the variable 'sample_mean_a'.")
print(f"Sample Mean: {sample_mean_a}")

The sample mean has already been calculated and stored in the variable 'sample_mean_a'.
Sample Mean: 1.15


**Reasoning**:
The subtask requires calculating the appropriate z-score for a 99% confidence interval in Scenario B, which assumes a known population standard deviation. This involves using the normal distribution (`norm`) from `scipy.stats`. The steps involve defining alpha, calculating the lower tail probability, and using the `ppf` method to find the z-score.



In [8]:
from scipy.stats import norm

# Define the significance level alpha
alpha = 1 - 0.99

# Calculate the lower tail probability
lower_tail_prob = alpha / 2

# Use the ppf method to find the z-score and take the absolute value
z_score_b = abs(norm.ppf(lower_tail_prob))

# Print the calculated z-score
print(f"Z-score for 99% confidence interval (Scenario B): {z_score_b}")

Z-score for 99% confidence interval (Scenario B): 2.5758293035489004


**Reasoning**:
I will now calculate the margin of error for Scenario B. This involves defining the known population standard deviation, calculating the standard error of the mean using this standard deviation and the previously calculated sample size, and then multiplying this standard error by the z-score to get the margin of error. Finally, I will print the standard error and margin of error for Scenario B.



In [9]:
population_std_dev_b = 0.2
standard_error_b = population_std_dev_b / math.sqrt(sample_size)
margin_of_error_b = z_score_b * standard_error_b

print(f"Standard Error (Scenario B): {standard_error_b}")
print(f"Margin of Error (Scenario B): {margin_of_error_b}")

Standard Error (Scenario B): 0.06324555320336758
Margin of Error (Scenario B): 0.16290974926039525


**Reasoning**:
Calculate the lower and upper bounds of the confidence interval for Scenario B and print the result.



In [10]:
lower_bound_b = sample_mean_a - margin_of_error_b
upper_bound_b = sample_mean_a + margin_of_error_b

print(f"99% Confidence Interval (Scenario B): [{lower_bound_b:.3f}, {upper_bound_b:.3f}]")

99% Confidence Interval (Scenario B): [0.987, 1.313]


## Present results

### Subtask:
Display the calculated confidence intervals for both scenarios and provide explanations for each.


**Reasoning**:
Display the calculated confidence intervals and provide explanations for both scenarios as requested by the instructions.



In [11]:
# Print results for Scenario A
print("--- Scenario A Results (Sample Std Dev, t-distribution) ---")
print(f"99% Confidence Interval (Scenario A): [{lower_bound_a:.3f}, {upper_bound_a:.3f}]")
print("\nExplanation for using the t-distribution in Scenario A:")
print("The t-distribution is used for constructing confidence intervals for the mean when the sample size is small AND the population standard deviation is unknown. In Scenario A, we have a small sample size (n=10) and we are using the sample standard deviation as an estimate because the population standard deviation is not provided or assumed to be unknown. Under these conditions, the t-distribution provides a more appropriate and conservative estimate for the confidence interval compared to the z-distribution, which assumes a known population standard deviation or a large sample size (typically n > 30). The t-distribution accounts for the additional uncertainty introduced by estimating the population standard deviation from the sample.")

# Print results for Scenario B
print("\n--- Scenario B Results (Known Population Std Dev, z-distribution) ---")
print(f"99% Confidence Interval (Scenario B): [{lower_bound_b:.3f}, {upper_bound_b:.3f}]")
print("\nExplanation for using the z-distribution in Scenario B:")
print("The z-distribution is used for constructing confidence intervals for the mean when the population standard deviation is known, regardless of the sample size (although it's more commonly applied with larger sample sizes when the population standard deviation is unknown, due to the Central Limit Theorem). In Scenario B, we are explicitly given a known population standard deviation of 0.2 million characters. This knowledge allows us to use the z-distribution, which is simpler and doesn't require accounting for the additional variability introduced by estimating the standard deviation from the sample.")

# Compare the intervals
print("\n--- Comparison of Confidence Intervals ---")
print(f"Scenario A Interval (t-dist): [{lower_bound_a:.3f}, {upper_bound_a:.3f}]")
print(f"Scenario B Interval (z-dist): [{lower_bound_b:.3f}, {upper_bound_b:.3f}]")
print("\nComparison:")
print("The confidence interval in Scenario A (using the t-distribution and sample standard deviation) is slightly narrower than the confidence interval in Scenario B (using the z-distribution and known population standard deviation). This difference in width is primarily due to the distribution used and the information available. The t-distribution is generally wider than the z-distribution for the same confidence level and a small sample size because it accounts for the increased uncertainty when the population standard deviation is unknown and estimated from the sample. With a known population standard deviation (Scenario B), there is less uncertainty, leading to a slightly narrower interval. The specific values of the sample standard deviation versus the known population standard deviation also influence the margin of error and thus the width.")

--- Scenario A Results (Sample Std Dev, t-distribution) ---
99% Confidence Interval (Scenario A): [1.039, 1.261]

Explanation for using the t-distribution in Scenario A:
The t-distribution is used for constructing confidence intervals for the mean when the sample size is small AND the population standard deviation is unknown. In Scenario A, we have a small sample size (n=10) and we are using the sample standard deviation as an estimate because the population standard deviation is not provided or assumed to be unknown. Under these conditions, the t-distribution provides a more appropriate and conservative estimate for the confidence interval compared to the z-distribution, which assumes a known population standard deviation or a large sample size (typically n > 30). The t-distribution accounts for the additional uncertainty introduced by estimating the population standard deviation from the sample.

--- Scenario B Results (Known Population Std Dev, z-distribution) ---
99% Confidence Int

## Summary:

### Data Analysis Key Findings

*   The sample mean durability is 1.15 million characters.
*   The sample standard deviation is approximately 0.108 million characters.
*   For a 99% confidence interval with a sample size of 10 and unknown population standard deviation (Scenario A), the critical t-score is approximately 3.250, and the confidence interval is \[1.039, 1.261] million characters.
*   For a 99% confidence interval with a known population standard deviation of 0.2 million characters (Scenario B), the critical z-score is approximately 2.576, and the confidence interval is \[0.987, 1.313] million characters.
*   The confidence interval calculated using the t-distribution and sample standard deviation (Scenario A) is slightly narrower than the interval calculated using the z-distribution and known population standard deviation (Scenario B).

### Insights or Next Steps

*   The difference in the width of the confidence intervals highlights the impact of knowing the population standard deviation versus estimating it from a small sample.
*   Further investigation could explore if the print-head durability data meets the assumption of normality, which is underlying for these confidence interval calculations.
